systemtest/src/test/java/io/strimzi/systemtest/RecoveryClusterTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest;
+
+import io.strimzi.test.ClusterController;
+import io.strimzi.test.EnvVariables;
+import io.strimzi.test.KafkaCluster;
+import io.strimzi.test.Namespace;
+import io.strimzi.test.StrimziRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.strimzi.test.k8s.BaseKubeClient.CM;
+import static io.strimzi.test.k8s.BaseKubeClient.DEPLOYMENT;
+import static io.strimzi.test.k8s.BaseKubeClient.SERVICE;
+import static io.strimzi.test.k8s.BaseKubeClient.STATEFUL_SET;
+
+@RunWith(StrimziRunner.class)
+@Namespace(RecoveryClusterTest.NAMESPACE)
+@ClusterController(envVariables = {
+    @EnvVariables(key = "STRIMZI_FULL_RECONCILIATION_INTERVAL_MS", value = "15000"),
+    @EnvVariables(key = "STRIMZI_OPERATION_TIMEOUT_MS", value = "15000")})
+@KafkaCluster(name = RecoveryClusterTest.CLUSTER_NAME, kafkaNodes = 1)
+public class RecoveryClusterTest extends AbstractClusterTest {
+
+    static final String NAMESPACE = "recovery-cluster-test";
+    static final String CLUSTER_NAME = "recovery-cluster";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecoveryClusterTest.class);
+
+    @Test
+    public void testDeleteTopicControllerDeployment() {
```

Comment:
We should try to name these tests better. We're not testing that we *can* delete the TC, we're testing that we can *recover* from TC deletion. So something like `testRecoveryFromTopicControllerDeletion()` would be clearer imho.

Reply:
Yeah, it sounds much more reasonable! Will rename class and methods.